src/panels/UserPanelsPanel/UserPanelsPanel.tsx

```diff
@@ -0,0 +1,159 @@
+      }
+
+      const root = await luaApi.absPath(UserPanelsFolderKey);
+      const folders = await luaApi.walkDirectoryFolders(root);
+
+      if (!folders) {
+        return;
+      }
+      const isWindowsMachine = navigator.userAgent.indexOf(WindowsKey) !== -1;
+      const slash = isWindowsMachine ? '\\' : '/';
+      const folderNames = Object.values(folders).map((panel) =>
+        // Get the folder name from the path
+        panel.substring(panel.lastIndexOf(slash) + 1)
+      );
+      dispatch(intializeUserPanels(folderNames));
+    };
+
+    if (luaApi && !isDataInitialized) {
+      getUserPanels();
+    }
+  }, [dispatch, isDataInitialized, luaApi]);
+
+  function addLocalPanel() {
+    if (!selectedPanel) {
+      return;
+    }
+    // TODO (ylvse) 2024-12-04: Change this back to window.location.host once we no longer
+    // serve the gui on port 4670 (the webpanels are served on port 4680)
+    // const src = `http://${window.location.host}/webpanels/${panelName}/index.html`;
+    const src = `http://localhost:4680/webpanels/${selectedPanel}/index.html`;
+
+    openPanel(src, selectedPanel);
+
+    setSelectedPanel(null);
+  }
+
+  function addWebPanel() {
+    if (!panelURL) {
+      return;
+    }
+    const startsWithHttp = panelURL.indexOf('http') === 0;
+    const src = startsWithHttp ? panelURL : `http://${panelURL}`;
+    const title = urlPanelTitle === '' ? src : urlPanelTitle;
+
+    openPanel(src, title);
+
+    setPanelURL('');
+    setUrlPanelTitle('');
+    dispatch(openWebpanel({ title: title, src: src }));
+  }
+
+  function openPanel(src: string, title: string) {
+    addWindow(<UserPanel src={src} title={title} />, {
+      title: title,
+      position: 'right',
+      id: title
+    });
+  }
+
+  return (
+    <Container>
+      <Title my={'xs'} order={2}>
+        User Panels
+      </Title>
+      <Title my={'xs'} order={3}>
+        Open Local Panel
+      </Title>
+      <Group align={'flex-end'}>
+        <Select
+          placeholder={'Select panel'}
+          data={localPanels}
+          onChange={setSelectedPanel}
+          value={selectedPanel}
+          flex={1}
+          onKeyDown={(e) => e.key === 'Enter' && addLocalPanel()}
+        />
+        <ActionIcon onClick={addLocalPanel} disabled={!selectedPanel} size={'lg'}>
+          <OpenIcon />
+        </ActionIcon>
+      </Group>
+      <Divider my={'md'} />
+      <Title order={3} my={'xs'}>
+        Open from URL
+      </Title>
+      <TextInput
+        value={urlPanelTitle}
+        label={'Title (optional)'}
+        placeholder={'Input title (optional)'}
+        onChange={(evt) => setUrlPanelTitle(evt.target.value)}
+      />
+      <Group align={'flex-end'} justify={'space-between'}>
+        <TextInput
+          value={panelURL}
+          label={'URL'}
+          placeholder={'Input URL'}
+          onChange={(evt) => setPanelURL(evt.target.value)}
+          onKeyDown={(e) => e.key === 'Enter' && addWebPanel()}
```

Comment:
I think we should try to be consistent with the event parameter, should we do `e`, `event`, or some other variant? This goes for all places in code